### PR TITLE
Integrate coverage

### PR DESCRIPTION
**Pull request by stephenfin**
This should help us identify dead code and other optimizations.

### CODE REVIEW

File: rst2pdf/tests/conftest.py [hunk 1]
```diff
@@ -43,8 +43,23 @@ def _get_pages(pdf):
...
 def compare_pdfs(path_a, path_b):
-    pdf_a = fitz.open(path_a)
-    pdf_b = fitz.open(path_b)
+    try:
+        pdf_a = fitz.open(path_a)
+    except RuntimeError:
+        pytest.fail(
+            'Reference file at %r is not a valid PDF.'
+            % (os.path.relpath(path_a, ROOT_DIR),),
+            pytrace=False,
```
Comment (akrabat) [inline rst2pdf/tests/conftest.py:52]: How do I test this? If I remove the Inkscape app from /Application, I get a long traceback starting with `File 'output/test_fancytitles.pdf' was not generated:`. Presumably there's something else I can do that created the PDF, but then causes PyMuPDF to fail?

Comment (stephenfin) [reply rst2pdf/tests/conftest.py:52]: I was seeing the issue due to a [broken wheel for `PIL`](https://github.com/python-pillow/Pillow/issues/4225) so I'm not sure how you could test with an actual broken package. If you wanted to test this artificially, you could insert some code before this to remove the generated file and replace it with e.g. an empty file or a text file?

**Review by akrabat — APPROVED** [review]

**Comment by akrabat** [comment]
It would be nice to know which tests cover which lines of code. Also, we have a lot of tests to write!

**Comment by akrabat** [comment]
Rebased to fix conflict

**Pull request closed without merging**